core/src/test/java/com/linecorp/armeria/server/ServerTest.java

```diff
@@ -358,6 +360,57 @@ public void customStartStopExecutor() {
         threads.forEach(t -> assertThat(t.getName()).startsWith(prefix));
     }
 
+    @Test
+    public void gracefulShutdownBlockingTaskExecutor() {
+        final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+        final Server server = new ServerBuilder()
+                .blockingTaskExecutor(executor, true)
+                .service("/", (ctx, req) -> HttpResponse.of(200))
+                .build();
+
+        server.start().thenApply(unused -> Thread.currentThread()).join();
```

Comment:
nit: `.thenApply(unused -> Thread.currentThread())` looks not necessary. Would you remove it?

Reply:
That is my mistake 😮, Thank you for thoughtful comments